### PR TITLE
build: Use eslint for jsx code

### PR DESCRIPTION
**Pull request by dperpeet**
We didn't use lint for our `*.jsx` files so far. Enabling this found a few cosmetic issues but also the odd bug.

As a rule we don't allow unused variables and using undefined variables.

I added the globals `require` and `module`.

This replaces work done in #5091, since that plugin was deprecated.

### CODE REVIEW

File: pkg/docker/storage.jsx [hunk 1]
```diff
@@ -417,7 +416,7 @@
         }
     }
 
-    function reset_storage(client, service) {
+    function reset_storage(client, service) { // eslint-disable-line no-unused-vars
```
Comment (dperpeet) [inline pkg/docker/storage.jsx]: @mvollmer do we need to keep this service parameter?

File: pkg/storaged/content-views.jsx [hunk 2]
```diff
@@ -594,7 +594,7 @@ function append_logical_volume_block(client, rows, level, block, lvol) {
             size: utils.fmt_size(block.Size),
             text: lvol.Name
         };
-        tabs = create_tabs(clienta, block, false);
+        tabs = create_tabs(client, block, false);
```
Comment (dperpeet) [inline pkg/storaged/content-views.jsx]: @mvollmer I assume this was a bug

File: pkg/docker/storage.jsx (hunk 1 above)
Comment (mvollmer) [inline pkg/docker/storage.jsx]: No.

File: pkg/storaged/content-views.jsx (hunk 2 above)
Comment (mvollmer) [inline pkg/storaged/content-views.jsx]: Yes.  This code path was not executed by the tests.  It only happens for logical volumes with partition tables.  I am not sure whether it is actually possible to partition lvs.

**Comment by dperpeet** [comment]
I may have to rebuild the test runners for this since they're using an older npm version (fedora 23).

**Comment by dperpeet** [comment]
even if the tests pass here, don't merge until @jscotka verifies this works downstream

**Comment by dperpeet** [comment]
The test runs are old, so we should trigger again once @jscotka has looked at this

**Comment by dperpeet** [comment]
Since this pull request changes `packages.json`, we currently can't backport this patch on top of a downstream release to test it, so @jscotka won't be able to look at it.

**Comment by dperpeet** [comment]
Rebased and fixed all current lint issues.

**Comment by dperpeet** [comment]
this will need a rebase when #5759 lands

**Comment by dperpeet** [comment]
rebased

**Pull request closed without merging**